sourcesapi/sourcesapi.py

```diff
@@ -0,0 +1,309 @@
+                                if type(val) is not bool:
+                                    logger.warning(module + " has incorrect key type: " + key + ":" + str(val))
+                                    errors_found = True
+
+                        for key, val in talker_info[module].source_details["settings_options"].items():
+                            if key == "enabled":
+                                if type(val["value"]) is not bool:
+                                    logger.warning(module + " has incorrect key type: " + key + ":" + str(val))
+                                    errors_found = True
+                            if key == "url_root":
+                                if not val["value"]:
+                                    logger.warning(module + " has missing value: " + key + ":" + str(val))
+                                    errors_found = True
+
+                        if errors_found:
+                            logger.warning(module + " is missing required settings. Check logs.")
+                            return False
+
+            del testmodule  # Needed?
+            if talker_info:
+                return talker_info
+            else:
+                return False
+
+        sources = list_sources()
+        talkers = {}
+        for source in sources:
+            talker_info = get_talker_info(source)
+            if talker_info:
+                talkers.update(talker_info)
+        if not talkers:
+            raise SourceInfoError("None", 4, "No viable comic information sources found! Check logs.")
+        else:
+            return talkers
+
+    # For issueidentifier
+    def set_log_func(self, log_func: Callable[[str], None]) -> None:
+        self.talker.log_func = log_func
+
+    def get_static_options(self) -> SourceStaticOptions:
+        return self.talker.source_details.static_options
+
+    def check_api_key(self, key: str, url: str, source_id: str):
+        for source in self.sources.values():
+            if source.source_details.id == source_id:
+                return source.check_api_key(key, url)
+        # Return false as back up or error?
+        return False
+
+    # Master function to search for series/volumes
+    def search_for_series(
+        self,
+        series_name: str,
+        callback: Callable[[int, int], None] | None = None,
+        refresh_cache: bool = False,
+        literal: bool = False,
+    ) -> list[ComicSearchResults]:
+        try:
+            series_result = self.talker.search_for_series(series_name, callback, refresh_cache, literal)
+            return series_result
+        except NotImplementedError:
+            logger.warning(self.talker.source_details.name + " has not implemented: " + "'search_for_series'")
+            raise SourceInfoError(
+                self.talker.source_details.name,
+                4,
+                "The source has not implemented: " + "'search_for_series'",
+            )
```

Comment:
You have some unnecessary string concatenation happening please remove.
Example:
```suggestion
        except NotImplementedError:
            logger.warning(f"{self.talker.source_details.name} has not implemented: 'search_for_series'")
            raise SourceInfoError(
                self.talker.source_details.name, 4, "The source has not implemented: 'search_for_series'"
            )
```